locale.c

```diff
@@ -441,6 +441,22 @@ S_wsetlocale(const int category, const wchar_t * wlocale)
 #  endif
 #endif  /* WIN32_USE_FAKE_OLD_MINGW_LOCALES */
 
+/* 'for' loop headers to hide the necessary casts */
+#define all_individual_category_indexes(i)                                  \
+    locale_category_index i = (locale_category_index) 0;                    \
+    i < LC_ALL_INDEX_;                                                      \
+    i = (locale_category_index) ((int) i + 1)
+
+#define all_but_0th_individual_category_indexes(i)                          \
+    locale_category_index i = (locale_category_index) 1;                    \
+    i < LC_ALL_INDEX_;                                                      \
+    i = (locale_category_index) ((int) i + 1)
+
+#define all_category_indexes(i)                                             \
+    locale_category_index i = (locale_category_index) 0;                    \
+    i <= LC_ALL_INDEX_;                                                     \
+    i = (locale_category_index) ((int) i + 1)
+
```

Comment:
This looks to be redundant with `all_individual_category_indexes`, and unused.

Reply:
Maybe the name should be changed to for_every_category.  The difference is that the upper limit is `<=` which includes `LC_ALL`, whereas the other includes only the individual categories, and the upper limit is `<`.  Future commits in the pipeline use it.

Reply:
I'll come up with a better name when the first use of this macro is added